osaca/semantics/hw_model.py

```diff
@@ -264,13 +403,16 @@ def get_data_ports(self):
     def get_full_instruction_name(instruction_form):
         """Get one instruction name string including the mnemonic and all operands."""
         operands = []
-        for op in instruction_form["operands"]:
-            op_attrs = [
-                y + ":" + str(op[y])
-                for y in list(filter(lambda x: True if x != "class" else False, op))
-            ]
-            operands.append("{}({})".format(op["class"], ",".join(op_attrs)))
-        return "{}  {}".format(instruction_form["name"].lower(), ",".join(operands))
+        for op in instruction_form.operands:
+            op_attrs = []
+            if op.name is not None:
+                op_attrs.append("name:" + op.name)
+            if op.prefix is not None:
+                op_attrs.append("prefix:" + op.prefix)
+            if op.shape is not None:
+                op_attrs.append("shape:" + op.shape)
+            operands.append("{}({})".format("register", ",".join(op_attrs)))
+        return "{}  {}".format(instruction_form.instruction.lower(), ",".join(operands))
```

Comment:
What about all other attributes besides `name`, `prefix`, and `shape`?

Reply:
After our discussion, this function should be removed and the tests should be moved to the `test_db_interface.py` file